app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DeleteRecordController.scala

```diff
@@ -16,32 +16,58 @@
 
 package uk.gov.hmrc.tradergoodsprofilesdatastore.controllers
 
+import org.apache.pekko.Done
 import play.api.Logging
-import play.api.mvc.{Action, AnyContent, ControllerComponents}
-import uk.gov.hmrc.http.UpstreamErrorResponse
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Request}
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 import uk.gov.hmrc.tradergoodsprofilesdatastore.connectors.RouterConnector
-import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.IdentifierAction
+import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.{IdentifierAction, StoreLatestAction}
+import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.RecordsRepository
 
 import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 class DeleteRecordController @Inject() (
   cc: ControllerComponents,
   routerConnector: RouterConnector,
-  identify: IdentifierAction
+  recordsRepository: RecordsRepository,
+  identify: IdentifierAction,
+  storeLatestAction: StoreLatestAction
 )(implicit ec: ExecutionContext)
     extends BackendController(cc)
     with Logging {
 
-  def deleteRecord(eori: String, recordId: String): Action[AnyContent] = identify.async { implicit request =>
-    routerConnector.deleteRecord(eori, recordId).map(response => response) transform {
-      case Success(_)                                                             => Success(NoContent)
-      case Failure(cause: UpstreamErrorResponse) if cause.statusCode == NOT_FOUND => Success(NotFound)
-      case Failure(cause: UpstreamErrorResponse)                                  =>
-        logger.error(s"Deleted record failed with ${cause.statusCode} with message: ${cause.message}")
-        Success(InternalServerError)
-    }
+  def deleteRecord(eori: String, recordId: String): Action[AnyContent] = (identify andThen storeLatestAction).async {
+    implicit request =>
+      isRecordActive(eori, recordId).flatMap {
```

Comment:
do we need to store the latest changes first here, what if it's outdated in our repository

Reply:
it's in the action line 42

Reply:
do we even need to go into our repository?

Reply:
yes - because that's how we check if the record is inactive - because the getRecord to the router (we don't currently have it) and it causes the same 400 issues anyway